From: Roarde <roarde@example.net>
Date: Wed 04 Nov 2020 07:08:58 AM UTC
Subject: [heirloom-sh] Allow Heirloom-specific profiles

If /etc/heirloom/profile exists, read it instead of /etc/profile; same
for an ~/.heirloom_profile and ~/.profile.

_____

diff -u ./defs.h ../new/defs.h
--- ./defs.h	2005-07-03 19:25:46.000000000 +0000
+++ ../new/defs.h	2020-10-30 19:13:00.980629847 +0000
@@ -461,7 +461,9 @@
 /* prompting */
 extern const char				stdprompt[];
 extern const char				supprompt[];
+extern const char				hlprofile[];
 extern const char				profile[];
+extern const char				syscfgdirs[];
 extern const char				sysprofile[];
 
 /* built in names */
Common subdirectories: ./fakewchar and ../new/fakewchar
diff -u ./main.c ../new/main.c
--- ./main.c	2005-07-03 19:25:46.000000000 +0000
+++ ../new/main.c	2020-10-30 19:55:43.016599989 +0000
@@ -268,14 +268,14 @@
 
 #ifndef RES
 
-			if ((input = pathopen(nullstr, sysprofile)) >= 0)
+			if ((input = pathopen(syscfgdirs, sysprofile)) >= 0)
 				exfile(rflag);		/* file exists */
 
 #endif
 			/* user profile */
 
-			if ((input = pathopen(homenod.namval, profile)) >= 0)
-			{
+			if ((input = pathopen(homenod.namval, hlprofile)) >= 0 ||
+			    (input = pathopen(homenod.namval, profile)) >= 0) {
 				exfile(rflag);
 				flags &= ~ttyflg;
 			}
diff -u ./msg.c ../new/msg.c
--- ./msg.c	2005-07-03 19:25:46.000000000 +0000
+++ ../new/msg.c	2020-10-30 19:16:40.050891092 +0000
@@ -151,8 +151,10 @@
 const char	readmsg[]	= "> ";
 const char	stdprompt[]	= "$ ";
 const char	supprompt[]	= "# ";
+const char	hlprofile[]	= ".heirloom_profile";
 const char	profile[]	= ".profile";
-const char	sysprofile[]	= "/etc/profile";
+const char	syscfgdirs[]	= "/etc/heirloom:/etc";
+const char	sysprofile[]	= "profile";
 
 /*
  * tables
